lib/elixir_internal_certificate/account/schemas/user.ex

```diff
@@ -0,0 +1,95 @@
+defmodule ElixirInternalCertificate.Account.Schemas.User do
+  use Ecto.Schema
+
+  import Ecto.Changeset
+
+  @email_regex ~r/^[^\s]+@[^\s]+$/
+  @email_max_length 160
+
+  @password_min_length 6
+  @password_max_length 72
+
+  @min_pole_of_byte 0
+
+  schema "users" do
+    field :email, :string
+    field :password, :string, virtual: true, redact: true
+    field :hashed_password, :string, redact: true
+
+    timestamps()
+  end
+
+  @doc """
+  A user changeset for registration.
+
+  It is important to validate the length of both email and password.
+  Otherwise databases may truncate the email without warnings, which
+  could lead to unpredictable or insecure behaviour. Long passwords may
+  also be very expensive to hash for certain algorithms.
+
+  ## Options
+
+    * `:hash_password` - Hashes the password so it can be stored securely
+      in the database and ensures the password field is cleared to prevent
+      leaks in the logs. If password hashing is not needed and clearing the
+      password field is not desired (like when using this changeset for
+      validations on a LiveView form), this option can be set to `false`.
+      Defaults to `true`.
+  """
+  def registration_changeset(user, attrs, opts \\ []) do
+    user
+    |> cast(attrs, [:email, :password])
+    |> validate_email()
+    |> validate_password(opts)
+  end
+
+  @doc """
+  Verifies the password.
+
+  If there is no user or the user doesn't have a password, we call
+  `Bcrypt.no_user_verify/0` to avoid timing attacks.
+  """
+  def valid_password?(
+        %__MODULE__{hashed_password: hashed_password},
+        password
+      )
+      when is_binary(hashed_password) and byte_size(password) > @min_pole_of_byte do
+    Bcrypt.verify_pass(password, hashed_password)
```

Comment:
> Prefer using the single-line function as long as mix format command satisfies to keep the function body just in one line.

Based on our convention, it will be one-liner syntax(instead of do end block)